src/Shared/BumpfeeModal.razor

```diff
@@ -0,0 +1,142 @@
+                                    MempoolRecommendedFeesType.EconomyFee => "Economy Fee",
+                                    MempoolRecommendedFeesType.HourFee => "Hour Fee",
+                                    MempoolRecommendedFeesType.HalfHourFee => "Half Hour Fee",
+                                    MempoolRecommendedFeesType.CustomFee => "Custom",
+                                    _ => lastFeeType.ToString()
+                                };
+                                <SelectItem Value="@lastFeeType">@lastFeeLabel</SelectItem>
+                            }
+                        </Select>
+                        <NumericPicker TValue="long"
+                                       Value="@(WithdrawalRequest != null ? (long)WithdrawalRequest.CustomFeeRate : 0)"
+                                       CurrencySymbolPlacement="CurrencySymbolPlacement.Suffix"
+                                       CurrencySymbol=" sats/vb"
+                                       Min="1"
+                                       Disabled="true">
+                        </NumericPicker>
+                    </div>
+                    <FieldHelp>@(_selectedMempoolRecommendedFeesType != MempoolRecommendedFeesType.CustomFee ? "Fees must be greater than the current one" : "")</FieldHelp>
+
+                </Field>
+            </Fields>
+        </ModalBody>
+        <ModalFooter>
+            <Button Color="Color.Secondary" Clicked="@HideModal">Cancel</Button>
+            <Button id="approve-button" Color="Color.Primary" Clicked="@HandleOnClick">
+                Approve
+            </Button>
+        </ModalFooter>
+    </ModalContent>
+    }
+</Modal>
+
+@inject IBitcoinService BitcoinService
+@inject IPriceConversionService PriceConversionService
+@inject INBXplorerService NBXplorerService
+@inject IWalletWithdrawalRequestRepository WalletWithdrawalRequestRepository
+@inject IWalletRepository WalletRepository
+@inject IFMUTXORepository FMUTXORepository
+@code {
+    [Inject]
+    public IToastService ToastService { get; set; }
+
+    [Parameter]
+    public ChannelOperationRequest? ChannelRequest { get; set; }
+    [Parameter]
+    public WalletWithdrawalRequest? WithdrawalRequest { get; set; }
+
+    [Parameter]
+    public Action<WalletWithdrawalRequest> SubmitBumpfeeModal { get; set; }
+
+    private decimal? _selectedWalletBalance;
+    private decimal _btcPrice;
+
+    private MempoolRecommendedFeesType _selectedMempoolRecommendedFeesType;
+    private long _customSatPerVbAmount = 1;
+
+    private Modal? _modalRef;
+
+    private async Task HandleOnClick()
+    {
+        var oldFee = WithdrawalRequest?.CustomFeeRate ?? 0;
+        if (_customSatPerVbAmount <= oldFee) {
+            ToastService.ShowError($"Fee must be greater than the current one ({oldFee} sats/vB)");
+        } else {
+            WalletWithdrawalRequest newRequest = new WalletWithdrawalRequest();
```

Comment:
are `WalletWithdrawalRequest` and `WithdrawalRequest` equal? maybe we could use the same class?